lib/utils.js

```diff
@@ -495,6 +495,16 @@ function applyWriteConcern(target, sources, options) {
   return target;
 }
 
+/**
+ * Checks if a given value is a Promise
+ *
+ * @param {*} maybePromise
+ * @return true if the provided value is a Promise
+ */
+function isPromiseLike(maybePromise) {
```

Comment:
we don't use this anymore, so we can delete it.

Reply:
yea let's just leave it in anyway for future use, no harm IMHO